encoding.go

```diff
@@ -91,6 +94,18 @@ func isDictionaryFormat(encoding format.Encoding) bool {
 	return encoding == format.PlainDictionary || encoding == format.RLEDictionary
 }
 
+func RegisterEncoding(enc encoding.Encoding) bool {
+	ns := encoding.NotSupported{}
+	if enc == ns {
+		return false
+	}
+	if LookupEncoding(enc.Encoding()) != ns {
+		return false
+	}
+	extraEncodings.Store(enc.Encoding(), enc)
+	return true
+}
```

Comment:
I'd prefer panicking here if the program attempts to override a default:
```suggestion
func RegisterEncoding(enc encoding.Encoding) {
	ns := encoding.NotSupported{}
	if enc == ns {
		panic("cannot register parquet encoding as not-supported")
	}
	if LookupEncoding(enc.Encoding()) != ns {
		panic("cannot register parquet encoding that overrides the standard specification")
	}
	extraEncodings.Store(enc.Encoding(), enc)
}
```